### PR TITLE
Fixes #227 - Add custom payload and pre-defined webhook.

### LINKED ISSUE #227: Add custom payload and pre-defined webhook
https://github.com/zammad/zammad/issues/4483
- Possibility to add the custom payload to a webhook (with code editor for JSON).

https://github.com/zammad/zammad/issues/4482
- Usage of pre-defined webhooks, which is currently available for:
  - Slack
  - Mattermost
  - RocketChat
(This should also include some hint of removing the article from the notification message)

```[tasklist]
### Tasks
- [x] Document custom payload for webhooks and the associated code editor
- [x] Document distinction of pre-defined webhooks and their fields/payloads
- [x] Add an example for MS Teams (not merged yet)
- [x] Convert Slack integration tutorial into a pre-defined webhook example and include a deprecation warning in the old section
- [x] Review
```

### PR DESCRIPTION
**Pull request by dvuckovic**
Changelog:

* Webhook section
  * Explanation about regular and pre-defined webhook types.
  * Hint about possibility to customize the payload (for both types of webhooks).
* Adding Webhooks section
  * Different ways to create a regular and pre-defined webhook, with screenshots.
  * Prominent warning that adding a webhook is not enough, that a corresponding Trigger or Scheduler must be added.
  * Cross-reference to HTTP Basic Auth fields in the existing warning.
  * Pre-defined webhook fields explanation.
  * Custom payload field explanation incl. short GIF with usage.
* Webhook Payload section
  * Default payload distinction between regular and pre-defined webhooks.
  * Added note about missing linked tickets in regular default payload.
* Webhook Examples section (NEW!)
  * Microsoft Teams Notifications tutorial, with a lot of screenshots.
  * Slack Notifications tutorial, even more screenshots!
  * Deprecation warning on the old Slack integration tutorial.

### CODE REVIEW

**Comment by MrGeneration** [comment]
Please make sure to blurr secrets also for URLs and even if the application no longer is configured / invalid.

**Comment by dvuckovic** [comment]
Thanks, but these were all _fake_ secrets anyway - no real systems were harmed in this production!

**Comment by MrGeneration** [comment]
> Thanks, but these were all fake secrets anyway - no real systems were harmed in this production!

Please do it anyway because you'll receive bug reports due to it.
Trust me. :)

**Comment by dvuckovic** [comment]
@dominikklein Ready for another review, Slack example has been added as discussed.

File: images/manage/webhook/webhook-slack-custom-payload-with-article-content.png
Comment (dominikklein) [inline images/manage/webhook/webhook-slack-custom-payload-with-article-content.png:1]: I think this overlapping content on the line numbers is currently a codemirror bug or?

**Review by dominikklein — APPROVED** [review]

**Comment by dominikklein** [comment]
@dvuckovic Looks good from my side (but I'm not a good candidate for textual QA :-D). Maybe we can create a follow up for creating a pre-defined example documentation for Mattermost + RocketChat.

**Comment by dvuckovic** [comment]
@mgruner Another review, from the wording point of view? Thanks!

**Comment by dvuckovic** [comment]
> @dvuckovic Looks good from my side (but I'm not a good candidate for textual QA :-D). Maybe we can create a follow up for creating a pre-defined example documentation for Mattermost + RocketChat.

https://github.com/zammad/zammad-admin-documentation/issues/231

File: images/manage/webhook/webhook-slack-custom-payload-with-article-content.png
Comment (MrGeneration) [reply images/manage/webhook/webhook-slack-custom-payload-with-article-content.png:1]: Overlapping content looks like a bug in my opinion? Can we overcome that issue? It looks broken like badly and I'd personally wouldn't want to see it documented that way. (Usually what I did in the past was to test run, open reports and wait for fixing) to reduce duplicate runs over documentary steps.

Comment (dvuckovic) [reply images/manage/webhook/webhook-slack-custom-payload-with-article-content.png:1]: This is our (Zammad) styling via custom CodeMirror theme, so not sure who else we could ask to "fix" this. I will try to revisit this soon, but I already fixed an issue recently related to these gutters when I switched to opaque color (there wasn't any comparable color for light mode).

**Review by mgruner — APPROVED** [review]
Nice! Some very minor textual suggestions. One sentence is broken and should be corrected.

**Comment by mgruner** [comment]
Great job @dvuckovic

**Pull request merged**